slamd/discovery/processing/experiment/experiment_preprocessor.py

```diff
@@ -0,0 +1,96 @@
+from slamd.common.error_handling import SequentialLearningException, ValueNotSupportedException, \
+    SlamdUnprocessableEntityException
+from slamd.discovery.processing.experiment.experiment_model import ExperimentModel
+
+
+class ExperimentPreprocessor:
+
+    @classmethod
+    def preprocess(cls, exp):
+        cls.filter_apriori_with_thresholds_and_update_orig_data(exp)
+        cls.filter_missing_inputs(exp)
+        cls.validate_experiment(exp)
+        cls.encode_categoricals(exp)
```

Comment:
filter_apriori_... modifies the rows of the dataframe, which could affect some of the validations. It needs to go first.
filter_missing_inputs currently drops columns which _currently_ does not affect the validations; however, in the future it might switch towards dropping rows in which case it needs to go before the validations.

The performance impact is negligible.